### PR TITLE
Fix timer recovery bug update

### PR DESCRIPTION
**Pull request by balrajsingh**

### CODE REVIEW

**Comment by yomimono** [comment]
Thanks very much for this patch, @balrajsingh .  Any chance there's an accompanying test you can post?  We've had many regressions in this part of the code over the last couple of years.

File: lib/tcp/segment.ml [hunk 1]
```diff
@@ -349,9 +350,9 @@ module Tx (Time:V1_LWT.TIME) (Clock:V1.MCLOCK) = struct
         | true ->
           q.dup_acks <- q.dup_acks + 1;
           if q.dup_acks = 3 ||
-             (q.dup_acks > 3 && Sequence.to_int32 ack_len > 0l) then begin
+            (Sequence.to_int32 ack_len > 0l) then begin
```
Comment (avsm) [inline lib/tcp/segment.ml]: I think this can be `Sequence.(gt ack_len zero)`

**Comment by jmitchell** [comment]
The CI build failed. Here are some errors I found in the [raw log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/158881143/log.txt):

```
mirage configure -f src/config.ml  -t unix
mirage: unknown option `-t'.
Usage: mirage configure [OPTION]... 
Try `mirage configure --help' or `mirage --help' for more information.
[31m[ERROR][m
  
ocamlfind ocamldep -package mirage -modules config.ml > config.ml.depends
ocamlfind ocamlc -c -annot -bin-annot -package mirage -o config.cmo config.ml
+ ocamlfind ocamlc -c -annot -bin-annot -package mirage -o config.cmo config.ml
File "config.ml", line 84, characters 46-51:
Error: Unbound value clock
Hint: Did you mean block, mclock or pclock?
Command exited with code 2.
Hint: Recursive traversal of subdirectories was not enabled for this build,
  as the working directory does not look like an ocamlbuild project (no
  '_tags' or 'myocamlbuild.ml' file). If you have modules in subdirectories,
  you should add the option "-r" or create an empty '_tags' file.
  
  To enable recursive traversal for some subdirectories only, you can use the
  following '_tags' file:
  
      true: -traverse
      <dir1> or <dir2>: traverse
      

make: *** [configure] Error 1
```

**Comment by yomimono** [comment]
#244 incorporating these changes has been merged, so closing this.

**Pull request closed without merging**